sdk/python/kfp/components/_python_op.py

```diff
@@ -535,6 +538,15 @@ def _func_to_component_spec_v2(
 
         arguments.extend(arguments_for_input)
 
+    # Add output placeholders for return values from func.
+    func_outputs = [
+        output for output in component_outputs
+        if output._passing_style is None
+    ]
+    for output in func_outputs:
```

Comment:
I think this should already been handled here:

https://github.com/kubeflow/pipelines/blob/46cd8196b0297b0af9966bd20dad4ddeac7142a5/sdk/python/kfp/components/_python_op.py#L535

If it's not, then there is some issue with io_types.OutputArtifact handling somewhere else.